Map voting

Apeami opened the pull request:
This pull request introduces a map voting system where players vote for the next map to be played. Four (can be changed) maps are selected from the available maps at random. After mode voting and before the map is loaded the players get to cast 1 vote by using the buttons. Map voting lasts for 20 seconds (can be changed). The map with the most votes is the one that is going to be played next. Ties are handled by selecting randomly. 

![image](https://github.com/user-attachments/assets/6fe87fa2-9d2f-4c48-9af1-c8d2b4c6238b)

AnonomousAccount commented:
Two things to add:

Firstly, if you could display a picture of each map to remind players of what map it is:
<img width="993" alt="Screenshot 2025-05-03 at 10 44 24 AM" src="https://github.com/user-attachments/assets/5e5cc296-55de-4e59-9f6b-32f2a62e3a70" />

Also, I would recommend changing the number of maps to three to slightly increase the randomness

Nanowolf4 commented:
> Two things to add:
> 
> Firstly, if you could display a picture of each map to remind players of what map it is: <img alt="Screenshot 2025-05-03 at 10 44 24 AM" width="993" src="https://private-user-images.githubusercontent.com/161108290/440136520-5e5cc296-55de-4e59-9f6b-32f2a62e3a70.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDYyOTcxMjgsIm5iZiI6MTc0NjI5NjgyOCwicGF0aCI6Ii8xNjExMDgyOTAvNDQwMTM2NTIwLTVlNWNjMjk2LTU1ZGUtNGU1OS05ZjZiLTMyZjJhNjJlM2E3MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUwM1QxODI3MDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NWE5YmY4OGMyMTk5YmJhY2FjODNjZDFmY2E4YTI4NzNiMjgxNDc0YTFkYjQ5YWMzMWE2ZDA2OWY3ODc5NDEwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.9WHCS6JM7-6Dz0QcWkVmYshJHM2ED6tZJ08xAKWnwO8">
> 
> Also, I would recommend changing the number of maps to three to slightly increase the randomness

This option is just beautiful!

Apeami commented:
I have changed the form-spec to add images of each map. I have re-adjusted the form-spec to put it into a horizontal layout. Do you think anything else needs to be added (map author, hint)?

![image](https://github.com/user-attachments/assets/6412ec50-45cd-4f4c-ad8b-7be0be4c2dcd)

AnonomousAccount commented:
Awesome! The only other thing I would think to add would be an 'abstain' button.

AnonomousAccount commented:
It _does_ show how many votes each map got, right?

AnonomousAccount commented:
Does it show the amount of votes for the losing maps aswell? I could not find that in the code, although I could just be blind.

Apeami commented:
I've added an Abstain button so players can choose not to vote for a map.
Also adjusted to send vote results of all maps, including those with zero votes, instead of only the ones that received at least one vote.

Nanowolf4 commented:
I believe this is ready for use. If any further issues/suggestions arise, I will be sure to let you know here.
Thank you for your contribution:)

Apeami commented:
Thanks for the review and for merging! I think it looks good as well and will be a great addition.
